### ISSUE TITLE
Model::getFrameSet() may not find all Frames

### ISSUE THREAD

**Issue by carmichaelong** [opensim-org/opensim-core#1461]
The description for `FrameSet` says the following:

> "List of Frames that various objects can be anchored to or expressed in, Body frames are built-in and not included in this list."

Using the `SinglePin.osim` model, `dumpSubcomponents()` prints the following:
```
Model 'SinglePin'
        Ground 'ground'
                Mesh 'ground_geom_0'
                FrameGeometry 'frame_geometry'
        Body 'rod1'
                Mesh 'rod1_geom_0'
                FrameGeometry 'frame_geometry'
        CustomJoint 'pin1'
                Coordinate 'r1_z'
                PhysicalOffsetFrame 'rod1_offset'
                        FrameGeometry 'frame_geometry'
```

Looping through `getBodySet()` returns the correct output:
```
printing body names, size = 1
Body rod1
```

Looping through `getFrameSet()` correctly does not list `Body rod1` (as per its own description) but it does not grab the `PhysicalOffsetFrame rod1_offset`:
```
printing frame names, size = 0
```

I don't think this behavior fits the description of `FrameSet`, even though `rod1_offset` is auto-generated. Should the comment be changed or the behavior?

**Comment by chrisdembia** [comment]
The FrameSet was created before it was possible to put any component within any other component. So at the time that the description was written, it was not ambiguous: such Frames couldn't really be anywhere else (they could be in the MiscModelComponents, but that wouldn't be the right place for them). I think the comment should be changed. I don't think it was ever intended that the FrameSet would be like the CoordinateSet, which is a non-owning bag of components that actually live elsewhere.

This is part of a larger discussion about what to do with Sets. And I don't remember what we decided.

**Comment by aseth1** [comment]
I agree the comment should be updated. I don't quite understand what "Body frames are built-in" means either.  I also don't see the need for the FrameSet since (as the example illustrates) Frames are generally in service of another Component (in this case a Joint), in which case it makes sense for it to live with that Component or related (e.g. Body) component. We don't have to shove everything into the top-level model and maybe we don't need a top-level FrameSet especially since it did not exist prior to 4.0 no one would miss it.

**Comment by jenhicks** [comment]
Some code in UpdateFromXML uses FrameSet. Ajay will look into changing that code and removing FrameSet completely. Also need to expose getFrameList.

**Comment by aseth1** [comment]
Closed by #1491
